### ISSUE TITLE
*tar files not untarred in extraction pipeline

### ISSUE THREAD

**Issue by AlexKurek** [mhardcastle/ddf-pipeline#290]
A new problem is that all 6 *.tar files are not untarred and Im getting:

```
Transferred:   	          0 B / 0 B, -, 0 B/s, ETA -
Checks:                 2 / 2, 100%
Elapsed time:         0.4s

[91m======================= Finding target name and position =======================[0m

Coordinate lookup gives ra=130.009733, dec=29.817425

[91m========================== Getting pointing positions ==========================[0m

We will use the following fields
    Field       ra    dec         sep        
                                  deg        
------------- ------ ----- ------------------
MS0839.8+2938 130.73 29.46 0.7208741710307898
      P130+32 130.71 31.91  2.177179197517556
      P129+29 129.86  29.4 0.4372534719774019

[91m=========================== Making working directory ===========================[0m


[91m============================ Downloading field data ============================[0m

[92mDoing field MS0839.8+2938[0m
Trying rclone download for field MS0839.8+2938
Macaroon maca_sksp_tape_DR2_readonly.conf does not exist!
Macaroon maca_sksp_tape_DDF.conf does not exist!
Running command rclone --multi-thread-streams 1 --config=/storage/kurek/maca_sksp_tape_DDF_readonly.conf listremotes
Running command rclone --multi-thread-streams 1 --config=/storage/kurek/maca_sksp_tape_DDF_readonly.conf -P --include-from /tmp/tmpdh9_idhm.txt copy maca_sksp_tape_DDF_readonly:other/MS0839.8+2938 /storage/kurek/4C29.30/4C29.30/MS0839.8+2938
Untarring files
[92mDoing field P130+32[0m
Initiating SDR download for field P130+32
Waiting for files to be online:
2/2... 
Downloading images.tar
Opening connection
Downloading 21365002240 bytes
Download successful, 21365002240 of 21365002240 bytes received in 1830.49 seconds (11.13 MB/s)
Downloading uv.tar
Opening connection
Downloading 134980392960 bytes
Download successful, 134980392960 of 134980392960 bytes received in 11561.00 seconds (11.13 MB/s)
Untarring files
[92mDoing field P129+29[0m
Initiating SDR download for field P129+29
Waiting for files to be online:
2/2... [94mToo few MS found for normal running: only 0[0m
Traceback (most recent call last):
  File "/usr/local/src/ddf-pipeline/scripts/sub-sources-outside-region.py", line 451, in <module>
    raise IOError('File', args['mslist'], 'does not exist and could not be created')
OSError: [Errno File] big-mslist.txt: 'does not exist and could not be created'
 - 23:53:01 - ClearSHM           | Clear shared memory
 - 23:53:01 - ClearSHM           | Clear Semaphores
 - 23:53:01 - ClearSHM           | Clear shared dictionaries

Downloading images.tar
Opening connection
Downloading 21369364480 bytes
Download successful, 21369364480 of 21369364480 bytes received in 1832.78 seconds (11.12 MB/s)
Downloading uv.tar
Opening connection
Downloading 130909900800 bytes
Download successful, 130909900800 of 130909900800 bytes received in 11217.01 seconds (11.13 MB/s)
Untarring files

[91m============================= Running subtraction  =============================[0m

[92mRunning: sub-sources-outside-region.py  --timeavg=1 --overwriteoutput --ncpu=28 -b ../4C29.30.ds9.reg -p 4C29.30[0m
[91mFAILED to run sub-sources-outside-region.py  --timeavg=1 --overwriteoutput --ncpu=28 -b ../4C29.30.ds9.reg -p 4C29.30: return value is 1[0m
Traceback (most recent call last):
  File "/usr/local/src/ddf-pipeline/scripts/extraction.py", line 115, in <module>
    run(executionstr,database=False)
  File "/usr/local/src/ddf-pipeline/utils/auxcodes.py", line 68, in run
    die('FAILED to run '+s+': return value is '+str(retval),database=database)
  File "/usr/local/src/ddf-pipeline/utils/auxcodes.py", line 51, in die
    raise RuntimeError(s)
RuntimeError: FAILED to run sub-sources-outside-region.py  --timeavg=1 --overwriteoutput --ncpu=28 -b ../4C29.30.ds9.reg -p 4C29.30: return value is 1
```

This is how it looks like:
`$ tree`
```
.
├── 4C29.30.ds9.reg
├── MS0839.8+2938
│   ├── images.tar
│   └── uv.tar
├── P129+29
│   ├── images.tar
│   └── uv.tar
└── P130+32
    ├── images.tar
    └── uv.tar
```

**Comment by mhardcastle** [comment]
OK, I will look at this. Let's not push any more changes to master until we have a recipe that actually works all the way to the end, please. This all worked fine before @twshimwell 's recent merge and it should be possible to get it to work again.

**Comment by twshimwell** [comment]
Ah oops. This is extractor.pay. Yes sorry I have broken this accidentally when tidying up the reprocessing things - you can see how it is in run_extraction_pipeline.py which works. I think easiest if we use the new prepare_field function in this loop instead of the do_sdr_and_rclone_download. Then that should do the untar and also fix up the solutions. 

for f in fields:
    field=f['Field']
    report('Doing field '+field)
    fdir=startdir+'/'+target+'/'+field
    if os.path.isdir(fdir):
        if len(glob.glob(fdir+'/*.ms.archive'))>0:
            warn('Field directory already contains MSs, skipping download')
            continue
do_sdr_and_rclone_download(field,fdir,verbose=True)

**Comment by mhardcastle** [comment]
OK, I'll take a look, and see if we can get #283 sorted at the same time.

**Comment by mhardcastle** [comment]
The version I have pushed to master now works for me -- or at least gets through to the subtraction step. It also fixes #283. @AlexKurek would you like to pull and test?

**Comment by AlexKurek** [comment]
I did, its working now.
About this new regions functionality, does it make sense to recompute all sources where extended emission was now visible because of artefacts?
